ssl/statem/statem_clnt.c

```diff
@@ -2454,7 +2450,6 @@ MSG_PROCESS_RETURN tls_process_new_session_ticket(SSL *s, PACKET *pkt)
     if (ticklen == 0)
         return MSG_PROCESS_CONTINUE_READING;
 
-    /* TODO(TLS1.3): Is this a suitable test for TLS1.3? */
```

Comment:
The comment below says that we enter this branch if we reused an existing session (I have some doubts as to whether this works as a test in <=TLSv1.2). But for TLSv1.3 at least it should work as intended. We don't have session_ids in TLSv1.3 but the code at the bottom of this function generates a synthetic one if we don't already have one. Thus in TLSv1.3 we will only take the branch below if the SSL_SESSION object has a ticket associated with it.

WRT 1-1 correspondence: SSL_SESSION objects have a different lifetime to tickets. They are created much earlier than the ticket arrives, to enable us to have somewhere to store the resumption information before the ticket has arrived.